python/pyspark/sql/tests.py

```diff
@@ -56,6 +56,15 @@
 from pyspark.sql.utils import AnalysisException, ParseException, IllegalArgumentException
 
 
+_have_arrow = False
+try:
+    import pyarrow
+    _have_arrow = True
```

Comment:
Maybe give the param doc string as exception message?

I.e., `To make use of Apache Arrow for conversion, pyarrow must be installed and available on the calling Python process (Experimental)`.